velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -165,15 +182,35 @@ ParquetRowReader::ParquetRowReader(
     memory::MemoryPool& pool)
     : reader_(std::move(reader)), pool_(pool) {
   auto& selector = *options.getSelector();
-  auto& filter = selector.getProjection();
   rowType_ = selector.buildSelectedReordered();
+  duckdbRowType_.reserve(rowType_->size());
+
+  auto& projection = selector.getProjection();
+  VELOX_CHECK_EQ(rowType_->size(), projection.size());
 
   std::vector<::duckdb::column_t> columnIds;
   columnIds.reserve(rowType_->size());
-  duckdbRowType_.reserve(rowType_->size());
-  for (auto& node : filter) {
-    columnIds.push_back(node.column);
-    duckdbRowType_.push_back(reader_->return_types[node.column]);
+  for (uint64_t i = 0; i < projection.size(); i++) {
+    uint64_t columnId = projection[i].column;
+    VELOX_CHECK_LT(
+        columnId,
+        reader_->names.size(),
+        "Unexpected column Id {} for name: {}",
```

Comment:
nit: no need to include column ID in the message as it will be included by the macro.